src/main/kotlin/com/jetbrains/micropython/nova/FileSystemWidget.kt

```diff
@@ -69,6 +69,25 @@ except Exception:
   
 """
 
+private val MPY_CALCULATE_CRC = { filePath: String ->
+    """
+import binascii
+
+try:
+    with open('$filePath', 'rb') as f:
+        crc = '%08x' % (binascii.crc32(f.read()) & 0xffffffff)
```

Comment:
As far as I understand, this code only calculates only 1st byte CRC, and this makes no sense

Reply:
f.read() reads the entire file into memory - actually it might be worth considering a chunk based approach instead of this one. But this CRC calculation should work.

Reply:
Then I am completely against this approach. It would work up to certain file size on "big" MCUs like ESP32, but will fail on small-memory devices like STM32 or ESP8266.

According to my research, it's possible to calculate crc chunk-by-chunk, but that is undocumented.
Moreover, I might take  mpremote  into use on communication level, and they promise crc support on the library level.
So probably it makes a little sense to implement crc right here and right now.

Reply:
Alright. I have removed the CRC checks from the code completely.